### PR TITLE
Feature - Favorite Card

### PR DESCRIPTION
**Pull request by agubler**
add-to-fav

fixes: #95

### CODE REVIEW

File: monster-cards/src/actions/widgetStoreActions.ts [hunk 1]
```diff
@@ -45,3 +46,28 @@ export const putCard = createAction({
...
+export const favCard = createAction({
+	do({ afterAll, puts, deletes }: ChangeRecord) {
+		if (puts.length) {
+			const item = puts[0];
+			widgetStore.put(assign({}, <any> item, {
+				type: 'fav-card',
+				classes: [ 'animated' ],
+				id: `fav-${item.id}`,
+				cardId: item.id
+			})).then(() => {
```
Comment (kitsonk) [inline monster-cards/src/actions/widgetStoreActions.ts]:
```
}))
.then(() => {
```

File: monster-cards/src/stores/favoriteStore.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	description: string;
+	score: number;
+	cardImage: string;
+	favouriteCount: number;
+}
+
+export type Store = MemoryStore<Card>;
+
+const favCardStore: Store = createMemoryStore<Card>({
+	data: [
```
Comment (kitsonk) [inline monster-cards/src/stores/favoriteStore.ts]: Just looks a bit odd, like something is missing...

File: monster-cards/src/stores/favoriteStore.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+export interface ChangeRecord {
+	beforeAll: Card[];
+	afterAll: Card[];
+	deletes: string[];
+	puts: Card[];
+}
+
+export function bindActions() {
+	return favCardStore.observe().subscribe((options: any) => {
```
Comment (kitsonk) [inline monster-cards/src/stores/favoriteStore.ts]: this seems odd... why not `options: ChangeRecord`?  Why the copy to another var and then do nothing with it?

If `options` can't be coerced to `ChangeRecord` in the parameters for some reason, why not:

```
return facCard.do(<ChangeRecord> options);
```

I just don't think it adds anything to the clarity of the code.

File: monster-cards/src/stores/favoriteStore.ts (hunk 2 above)
Comment (agubler) [inline monster-cards/src/stores/favoriteStore.ts]: agreed, was on the list to tidy

File: monster-cards/src/main.ts [hunk 4]
```diff
@@ -2,38 +2,94 @@ import createApp from 'dojo-app/createApp';
...
+			factory: createSearchInput
+		},
+		{
+			id: 'navbar-favorites',
+			factory: createContainer.mixin(createVNodeEvented),
+			options: {
+				tagName: 'li'
+			},
+			listeners: {
+				mouseover: () => { showFavorites.do(); },
```
Comment (agubler) [inline monster-cards/src/main.ts]: That's not quite right, but it can be `mouseover: showFavorites` (minus the `.do`)

**Comment by agubler** [comment]
Closing this feature as it is significantly outdated now.

**Pull request closed without merging**